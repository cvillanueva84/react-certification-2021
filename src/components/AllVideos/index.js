export { default } from './AllVideos.component';