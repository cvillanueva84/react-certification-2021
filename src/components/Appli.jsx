import React from 'react';
import CardList from './CardList';
import NavBar from './NavBar';

export default function Appli() {
  return (
    <div>
      <NavBar />
      <CardList />
    </div>
  );
}
