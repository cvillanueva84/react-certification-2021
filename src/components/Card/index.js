export {default} from './Card.component.jsx'; 