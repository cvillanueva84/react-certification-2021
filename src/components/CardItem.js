import React from 'react';
import { useContext } from 'react';
import { GlobalContext } from './Context';
import {CardItemUi}  from './ui/CardItemUi'

export const CardItem = ({ title, description, url, videoId }) => {

  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClick = () => {
      dispatch({
        type: 'actionAddVideoData',
        payload: {
          ...myStateReducer,
          videoData: {
            title: title,
            description: description,
            url: url,
            videoId: videoId
          }
        }
      });
  }

  return (
    <CardItemUi 
      title={title} 
      description={description} 
      url={url} 
      handleClick={handleClick}
    />
  );

};

