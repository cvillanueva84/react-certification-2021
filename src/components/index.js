export { default as Navbar } from './Navbar/Navbar';
export { default as SearchBar } from './SearchBar/SearchBar';
export { default as ThemeSwitch } from './ThemeSwitch/ThemeSwitch';
export { default as Results } from './Results/Results';
