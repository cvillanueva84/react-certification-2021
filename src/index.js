import React from 'react';
import ReactDOM from 'react-dom';
import App from './components/App';
import StoreProvider from './state/Store.provider';

ReactDOM.render(
  <React.StrictMode>
    <StoreProvider>
      <App />
    </StoreProvider>
  </React.StrictMode>,
  document.getElementById('root')
);
