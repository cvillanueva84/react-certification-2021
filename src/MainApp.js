import React, { useEffect, useReducer } from 'react'
import { AppRouter } from './routers/AppRouter'
import { GlobalContext } from './components/Context'
import { globalReducer } from './components/globalReducer';
import { ThemeProvider } from 'styled-components';
import { GlobalStyles, lightTheme, darkTheme } from './style/MainAppStyled';


const init = () => {
  return JSON.parse(localStorage.getItem('user')) || {
    mode: 'light',
    search: 'wizeline',
    videoData: {},
    videoDataFav: {},
    user: '',
    logged: false,
    favorites: [],
  }
}

export const MainApp = () => {

  const [myStateReducer, dispatch] = useReducer(globalReducer, {}, init);

  useEffect(() => {
    localStorage.setItem('user', JSON.stringify(myStateReducer));
  }, [myStateReducer]);

  return (
    <GlobalContext.Provider value={{myStateReducer, dispatch,} }>
      <ThemeProvider theme={myStateReducer.mode === 'light' ? lightTheme : darkTheme}>
        <GlobalStyles />
        <AppRouter />
      </ThemeProvider>
    </GlobalContext.Provider>
  )
}
