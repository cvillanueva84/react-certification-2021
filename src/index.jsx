import React from 'react';
import ReactDOM from 'react-dom';

import Appli from './components/Appli';
import './index.css';

ReactDOM.render(
  <React.StrictMode>
    <Appli />
  </React.StrictMode>,
  document.getElementById('root')
);
