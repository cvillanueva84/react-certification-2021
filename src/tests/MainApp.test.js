import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from "react-router";
import { GlobalContext } from "../components/Context";
import { MainApp } from "../../src/MainApp";


describe('Test MainApp', () => {
    
    afterEach(() => {
        jest.clearAllMocks();
    })

    it('should render success lightTheme', () => {
        const contextValue = {
            dispatch: jest.fn(),
            myStateReducer: {
                mode: 'light',
                search: 'wizeline',
                videoData: {}
            }
        }
        const historyMock = {
            push: jest.fn(),
            replace: jest.fn(),
            location: {},
            listen: jest.fn(),
            createHref: jest.fn()
        }
    
        const wrapper = mount(
            <GlobalContext.Provider value={contextValue}>
                <MemoryRouter>
                    <Router history={historyMock}>
                        <MainApp/>
                    </Router>
                </MemoryRouter>
            </GlobalContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });

    it('should render success darkTheme', () => {
        const contextValue = {
            dispatch: {
                mode: 'dark',
            },
            myStateReducer: {
                mode: 'dark',
                search: 'wizeline',
                videoData: {}
            }
        }
        const historyMock = {
            push: jest.fn(),
            replace: jest.fn(),
            location: {},
            listen: jest.fn(),
            createHref: jest.fn()
        }
    
        const wrapper = mount(
            <GlobalContext.Provider value={contextValue}>
                <MemoryRouter>
                    <Router history={historyMock}>
                        <MainApp/>
                    </Router>
                </MemoryRouter>
            </GlobalContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });
});