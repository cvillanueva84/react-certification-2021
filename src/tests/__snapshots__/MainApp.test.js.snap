// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MainApp should render success darkTheme 1`] = `ReactWrapper {}`;

exports[`Test MainApp should render success lightTheme 1`] = `ReactWrapper {}`;
